src/Engine/Objects/SpriteObject.cpp

```diff
@@ -762,32 +762,35 @@ void SpriteObject::ExplosionTraps() {
     }
     unsigned int v10 = ((unsigned int)(11 * dir_y) >> 5) + (dir_z / 4) + dir_x;
     if (v10 <= 768) {
-        int v11 = 5;
-        if (pMapInfo->Trap_D20)
-            v11 += grng->RandomDice(pMapInfo->Trap_D20, 20);
+        int trapDamage = 5;
+        if (pMapInfo->Trap_D20) {
+            trapDamage += grng->RandomDice(pMapInfo->Trap_D20, 20);
+        }
         DAMAGE_TYPE pDamageType;
         switch (this->uType) {
-            case 811:
+            case SPRITE_TRAP_FIRE:
                 pDamageType = DMGT_FIRE;
                 break;
-            case 812:
+            case SPRITE_TRAP_LIGHTNING:
                 pDamageType = DMGT_ELECTR;
                 break;
-            case 813:
+            case SPRITE_TRAP_COLD:
                 pDamageType = DMGT_COLD;
                 break;
-            case 814:
+            case SPRITE_TRAP_BODY:
                 pDamageType = DMGT_BODY;
                 break;
             default:
                 return;
         }
-        for (unsigned int i = 1; i <= 4; ++i) {
-            int v13 = pPlayers[i]->GetPerception() + 20;
-            if (pPlayers[i]->CanAct() && (grng->Random(v13) > 20))
-                pPlayers[i]->PlaySound(SPEECH_AvoidDamage, 0);
-            else
-                pPlayers[i]->ReceiveDamage(v11, pDamageType);
+        for (Player &player : pParty->pPlayers) {
+            int perceptionCheckValue = player.GetPerception() + 20;
+            // TODO(Nik-RE-dev): character that is asleep will not receive damage?
```

Comment:
CanAct returns false for sleeping characters, so if character is sleeping he'll be getting damaged.

Reply:
OK I'll merge and you can deal with this in a separate PR.